airflow-core/src/airflow/api_fastapi/common/exceptions.py

```diff
@@ -80,6 +82,24 @@ def _is_dialect_matched(self, exc: IntegrityError) -> bool:
         return False
 
 
+class DAGErrorHandler(BaseErrorHandler[DeserializationError]):
+    """Handler for DAG-related errors."""
+
+    def __init__(self):
+        super().__init__(DeserializationError)
+
+    def exception_handler(self, request: Request, exc: DeserializationError):
+        """Handle DAG deserialization exceptions."""
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"An error occurred while trying to deserialize DAG: {exc}",
+        )
+
+
 DatabaseErrorHandlers = [
     _UniqueConstraintErrorHandler(),
 ]
+
+DAGErrorHandlers = [
+    DAGErrorHandler(),
+]
```

Comment:
Do we really need these to be separate? How about just have one list called `ERROR_HANDLERS` instead.

Or, do we really need the list at all? Why not just import the classes directly in `app.py`?

Reply:
+1 unique list of `ErrorHandlers` seems more appropriate for now as there is really just one db handler.